### ISSUE TITLE
zig fails to chain usingnamespace

### ISSUE THREAD

**Issue by fengb** [ziglang/zig#2747]
I haven't narrowed down the trigger yet, but it happens when trying to test https://github.com/Hejsil/zig-bench/blob/master/bench.zig:

```
zig-bench$ zig build test
/usr/local/Cellar/zig/0.4.0+de2b0cd7/lib/zig/std/debug.zig:1048:23: error: container 'std.c' has no member called '_mh_execute_header'
    const hdr = &std.c._mh_execute_header;
                      ^
/usr/local/Cellar/zig/0.4.0+de2b0cd7/lib/zig/std/c.zig:46:29: error: use of undeclared identifier 'fd_t'
pub extern "c" fn write(fd: fd_t, buf: [*]const u8, nbyte: usize) isize;
                            ^
/usr/local/Cellar/zig/0.4.0+de2b0cd7/lib/zig/std/c.zig:18:30: error: use of undeclared identifier '_errno'
        return @intCast(u16, _errno().*);
                             ^
/usr/local/Cellar/zig/0.4.0+de2b0cd7/lib/zig/std/os.zig:1990:18: note: called from here
    switch (errno(system.munmap(memory.ptr, memory.len))) {
                 ^
```

**Comment by fengb** [comment]
I managed to squelch the 2nd error (`fd_t`) by tossing in an extra `usingnamespace`:

```
pub usingnamespace @import("os/bits.zig");
usingnamespace @import("os/bits.zig");
```

**Comment by mikdusan** [comment]
and this hack somehow helps:

```diff
diff --git a/bench.zig b/bench.zig
index 9c95fc4..67c6b01 100644
--- a/bench.zig
+++ b/bench.zig
@@ -14,6 +14,7 @@ pub fn benchmark(comptime B: type) !void {
     const args = if (@hasDecl(B, "args")) B.args else [_]void{{}};
     const iterations: u32 = if (@hasDecl(B, "iterations")) B.iterations else 100000;
 
+    const x = std.c._mh_execute_header;
     comptime var max_fn_name_len = 0;
     const functions = comptime blk: {
         var res: []const Decl = [_]Decl{};
```

**Comment by fengb** [comment]
Much more targeted example: https://github.com/fengb/zig-usingnamespace-bug

main.zig's `usingnamespace @import("outer.zig");` does not trigger outer.zig's `pub usingnamespace @import("inner.zig");`

Removing the `pub` keyword or explicitly referencing a variable within inner.zig will force the chained namespace into existence.

**Comment by andrewrk** [comment]
Thanks for this report. This is a nasty bug, high priority to fix.

**Comment by emekoi** [comment]
possible duplicate: #2632

**Comment by andrewrk** [comment]
I suspect it's also related to #2580

**Comment by emekoi** [comment]
@andrewrk closed in af86614?

**Comment by fengb** [comment]
All of the previous issues are now fixed! 👍
